### PR TITLE
✨ NEW: Add @unified-myst/core-parse

### PR DESCRIPTION
**Pull request by chrisjsewell**

### CODE REVIEW

File: packages/process-roles-directives/src/index.js [hunk 1]
```diff
@@ -42,16 +54,16 @@ export function processRolesDirectives(
     /** @param {Node} node */
     function collectIdentifiers(node) {
         switch (node.type) {
-            case 'paragraph' || 'heading' || 'mystRole' || 'mystDirective':
+            case 'paragraph' || 'heading' || 'role' || 'mystDirective':
```
Comment (rowanc1) [inline packages/process-roles-directives/src/index.js:57]: Should be mystRole / mystDirective?

File: packages/core-parse/README.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+- Minimise AST walks:
+  - Concept of transforms that are purely data collectors
+    - Then they can be run at the same time, rather than performing multiple AST walks
+    - Maybe change the signature of `afterTransforms`, so that it is called on a single walk through the AST (for every node)
+
+- Disabling extensions, and even specific directives/roles/transforms within an extension.
+
+- How to handle conversion to output formats?
+  - For HTML, basically we want for extensions to be able to supply <https://github.com/syntax-tree/mdast-util-to-hast#optionshandlers>, and this would likely be similar for other formats
+  - But do we also include `mdast-util-to-hast` as a dependency here, since this would not be good for package size when it it not used?
```
Comment (rowanc1) [inline packages/core-parse/README.md:249]: Very much agree this shouldn't be included, or is a separate lib.

File: packages/core-parse/README.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+//   "core": {
+//     "cname1": "test",
+//     "cname2": []
+//   }
+// }
+```
+
+## Design decisions
+
+The design is intended to quite closely mirror that of [docutils](https://docutils.sourceforge.io) and [Sphinx](https://www.sphinx-doc.org).
```
Comment (rowanc1) [inline packages/core-parse/README.md:206]: Would love to see some mention that this is moving to support scientific writing, and that decisions will be made to change/improve underlying pieces to go in that direction.

File: packages/core-parse/README.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        const table = toc(ast)
+        ast.children.unshift(table.map)
+    }
+}
+
+myExtension = {
+  name: 'myExtension',
+  roles: { abbr: { processor: RoleAbbr } },
+  directives: { note: { processor: DirectiveNote } },
+  hooks: { afterRead: { addtoc: { priority: 100, processor: addToc } } },
```
Comment (rowanc1) [inline packages/core-parse/README.md:140]: What is priority here?

File: packages/core-parse/src/extensions/spans.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+/** Taken from https://github.com/live-clones/docutils/blob/48bb76093b4ba83654b2f2c86e7c52c4bb39c63b/docutils/docutils/parsers/rst/roles.py#L257-L264 */
+for (const role of [
+    'abbreviation',
+    'acronym',
+    'literal',
+    'emphasis',
+    'strong',
+    'subscript',
+    'superscript',
+    'title-reference',
```
Comment (rowanc1) [inline packages/core-parse/src/extensions/spans.js:21]: What is this, can we remove?

File: packages/core-parse/src/extensions/spans.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
+/** Roles for wrapping inline text in containers with semantic meaning. */
+import { u } from 'unist-builder'
+
+import { RoleProcessor } from '../roleProcessor.js'
+
+export const spansExtension = {
+    name: 'spans',
```
Comment (rowanc1) [inline packages/core-parse/src/extensions/spans.js:7]: `inlineMarkup` extensions? Calling this span is strange to me, that means `<span>`, which isn't what these are? flowContent?

File: packages/process-roles-directives/package.json [hunk 7]
```diff
@@ -49,6 +49,7 @@
         "test": "node --experimental-vm-modules ../../node_modules/jest/bin/jest.js"
     },
     "dependencies": {
+        "micromark-util-normalize-identifier": "^1.0.0",
```
Comment (rowanc1) [inline packages/process-roles-directives/package.json:52]: @fwkoch this is good to know about.

**Review by rowanc1 — COMMENTED** [review]
Comments as I was looking through.

File: packages/core-parse/src/extensions/spans.js (hunk 5 above)
Comment (chrisjsewell) [reply packages/core-parse/src/extensions/spans.js:21]: https://docutils.sourceforge.io/docs/ref/rst/roles.html#title-reference

File: packages/core-parse/src/extensions/spans.js (hunk 6 above)
Comment (chrisjsewell) [reply packages/core-parse/src/extensions/spans.js:7]: Yeh I literally just made this name up before our call, and happy to change. `flowContent` doesn't seem very "human intuitive" though 🤔

File: packages/process-roles-directives/src/index.js (hunk 1 above)
Comment (chrisjsewell) [reply packages/process-roles-directives/src/index.js:57]: Ideally, these names would be referenced by some central constant

File: packages/core-parse/README.md (hunk 3 above)
Comment (chrisjsewell) [reply packages/core-parse/README.md:206]: Yep makes sense 👍

File: packages/core-parse/README.md (hunk 4 above)
Comment (rowanc1) [reply packages/core-parse/README.md:140]: Thanks.

File: packages/core-parse/src/extensions/spans.js (hunk 5 above)
Comment (rowanc1) [reply packages/core-parse/src/extensions/spans.js:21]: I feel like in the same way that we and not bringing everything over (e.g. pep-reference), this is probably better handled by a `cite` role.

File: packages/core-parse/src/extensions/spans.js (hunk 6 above)
Comment (rowanc1) [reply packages/core-parse/src/extensions/spans.js:7]: What about `inline` and `inlineExtensions` -- seems a bit general, but better than span?

Comment (chrisjsewell) [reply packages/core-parse/src/extensions/spans.js]: yeh, renamed to `inlineMarkup` 👍

File: packages/core-parse/src/extensions/spans.js (hunk 5 above)
Comment (chrisjsewell) [reply packages/core-parse/src/extensions/spans.js]: removed 👍

File: packages/core-parse/README.md (hunk 3 above)
Comment (chrisjsewell) [reply packages/core-parse/README.md:208]: added a line

**Comment by chrisjsewell** [comment]
Ok the general structure is all there now, so will merge

**Comment by chrisjsewell** [comment]
,

**Pull request closed without merging**